client/js/controllers/pool_controller.js

```diff
@@ -0,0 +1,149 @@
+
+            this._name = ctx.parameters.name;
+            pool.addEventListener('change', e => this._evtSaved(e, section));
+
+            const categories = {};
+            for (let category of poolCategoriesResponse.results) {
+                categories[category.name] = category.name;
+            }
+
+            this._view = new PoolView({
+                pool: pool,
+                section: section,
+                canEditAnything: api.hasPrivilege('pools:edit'),
+                canEditNames: api.hasPrivilege('pools:edit:names'),
+                canEditCategory: api.hasPrivilege('pools:edit:category'),
+                canEditDescription: api.hasPrivilege('pools:edit:description'),
+                canEditPosts: api.hasPrivilege('pools:edit:posts'),
+                canMerge: api.hasPrivilege('pools:merge'),
+                canDelete: api.hasPrivilege('pools:delete'),
+                categories: categories,
+                escapeColons: uri.escapeColons,
+            });
+
+            this._view.addEventListener('change', e => this._evtChange(e));
+            this._view.addEventListener('submit', e => this._evtUpdate(e));
+            this._view.addEventListener('merge', e => this._evtMerge(e));
+            this._view.addEventListener('delete', e => this._evtDelete(e));
+        },
+                  error => {
+            this._view = new EmptyView();
+            this._view.showError(error.message);
+        });
+    }
+
+    _evtChange(e) {
+        misc.enableExitConfirmation();
+    }
+
+    _evtSaved(e, section) {
+        misc.disableExitConfirmation();
+        if (this._name !== e.detail.pool.names[0]) {
+            router.replace(
+              uri.formatClientLink('pool', e.detail.pool.id, section),
+              null, false);
+        }
+    }
+
+    _evtUpdate(e) {
+        this._view.clearMessages();
+        this._view.disableForm();
+        if (e.detail.names !== undefined) {
+            e.detail.pool.names = e.detail.names;
+        }
+        if (e.detail.category !== undefined) {
+            e.detail.pool.category = e.detail.category;
+        }
+        if (e.detail.description !== undefined) {
+            e.detail.pool.description = e.detail.description;
+        }
+        if (e.detail.posts !== undefined) {
+            e.detail.pool.posts.clear()
+            for (let post_id of e.detail.posts) {
+                e.detail.pool.posts.add(Post.fromResponse({ id: parseInt(post_id) }))
+            }
+        }
+        e.detail.pool.save().then(() => {
+            this._view.showSuccess('Pool saved.');
+            this._view.enableForm();
+        }, error => {
+            this._view.showError(error.message);
+            this._view.enableForm();
+        });
+    }
+
+    _evtMerge(e) {
+        this._view.clearMessages();
+        this._view.disableForm();
+        e.detail.pool
+            .merge(e.detail.targetPoolId, e.detail.addAlias)
```

Comment:
`addAlias`? shouldn't there be a dedicated control to it?

Reply:
I used the same structure as tags, where aliases don't have a dedicated controller.